test/integration/good/warning/unreachable_statement.stan

```diff
@@ -1,6 +1,6 @@
 functions {
    void foo(real x){
-     if(x) ; // bug - misplaced ; makes the next statement unconditional
+     if(x == 0.0) ; // bug - misplaced ; makes the next statement unconditional
```

Comment:
I agree it shouldn't be this test necessarily, but we should add something to the deprecated_syntax test in this same `warning` folder which triggers all the new warnings.